### PR TITLE
[WIP] Re-work web ui architecture

### PR DESCRIPTION
**Pull request by langerhans**
**Fixed issue:** Mostly not a fix, but still added a fix for #534 

**Changes made:**

* Introduced Spark as lightweight web framework to serve the GUI and maybe more later (REST API?)
* Added trainer name to page title

**To dos (Yet this can be merged on its own)**

- [ ] Split up the huge HTML into the template, JS and CSS file for reusability.
    - [ ] Introduce a real templating engine
    - [ ] Remove the String replacements with actual models
    - [ ] Move all assets to respective folders
- [ ] Migrate to Spark's Jetty WebSocket

**Discussion about To dos**
About the templating: There are tons of options for a templating engine. Spark supports several out of the box, see a list here: http://sparkjava.com/documentation.html#views-templates
I personally have used Pebble before, which worked fine. Thymeleaf comes with the benefit that previews still render as normal HTML but the learning curve is steep. Velocity is recommended often and very mature.
To be honest, we probably won't need that many features for now, but we should keep the options open if the web gui gets bigger.

About the websocket: Spark comes with Jetty which comes with a websocket server. Problem is we use Socket.IO right now which is kinda convenient but also a bit special, so not easy to adapt. The only real advantage of moving would be one less dependency and being able to use the same port as the website is running on also for the websocket. But not sure if that's worth it.

### CODE REVIEW

**Comment by apottere** [comment]
Do we need backend templates?  This seems like a very good fit for a SPA.

File: src/main/resources/ink/abb/pogo/scraper/gui/index.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
 <html lang="en">
 <head>
     <meta charset="UTF-8">
-    <title>Pokemon Go Bot GUI</title>
+    <title>{{username}} - Pokemon Go Bot GUI</title>
```
Comment (shilch) [inline src/main/resources/ink/abb/pogo/scraper/gui/index.html]: You can change the title with JavaScript: `document.title = "Page title" `

Comment (langerhans) [inline src/main/resources/ink/abb/pogo/scraper/gui/index.html]: Yeah I know, but the nick will not change dynamically I think?

Comment (shilch) [inline src/main/resources/ink/abb/pogo/scraper/gui/index.html]: Yes, it will not change but data should be added in the client and not Server (my opinion)

**Comment by langerhans** [comment]
Closing in favor of #622

**Pull request closed without merging**